private/bufpkg/bufimage/util.go

```diff
@@ -368,6 +368,10 @@ func fileDescriptorProtoToProtoImageFile(
 			ModuleInfo:          protoModuleInfo,
 		},
 	}
+	// TODO: What if file descriptor's unknown fields contain the image file's buf_extension?
+	//       We should probably examine the unknown fields and discard field 8042 if found.
```

Comment:
Curious what you think about this. Should I go ahead and address this now? Do you think this is a material risk?

Reply:
Hmmm. I don't think its a material risk, I purposefully chose a borderline-nonsensical large number to make sure there would not be any overlap. If it's easy to address (<10 min), might be worth doing just to get it done though.

Reply:
Ok, I will go ahead and address this.

FWIW, I wasn't worried about a conflicting field being added to `FileDescriptorProto`. The risk I was thinking of was a scenario like so:
1. An `ImageFile` is serialized to a file.
2. That file gets deserialized into a `FileDescriptorProto`. (Like if we tell buf that a file is a file descriptor set when it is actually an image.)
3. The resulting `FileDescriptorProto` is then converted back to an `ImageFile`.

In a case like this, it would be possible for field 8042 to be present _both_ as a known field and an unknown field.

Reply:
Makes sense. Yea if we don't address this now, it's going to be a nasty bug someday.

Reply:
Definitely more than 10 minutes to address. But also definitely more robust/correct now. PTAL.

Reply:
should this todo be deleted now?